### PR TITLE
Categories crud

### PR DESCRIPTION
**Pull request by AndrewTehan**
# Description

[category CRUD](https://trello.com/c/g7LJcbkX)
## Type of change

- [x] Bug fix
- [x] New feature 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/api/v1/admin/categories.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          present categories, with: Entities::Category
+        end
+
+        desc 'Return a specific category'
+        route_param :id do
+          get do
+            category = Category.find(params[:id])
+            present category, with: Entities::Category
+          rescue ActiveRecord::RecordNotFound
+            error!({ error_code: 404, error_message: 'Category not found' })
```
Comment (hannazspot) [inline app/api/v1/admin/categories.rb]: We decided to separate this kind of logic in some general class/module and do it [in the scope of this ticket](https://trello.com/c/wGRbj1hx/70-add-error-handling-for-crud-operations)

Comment (hannazspot) [reply app/api/v1/admin/categories.rb]: That's why I'd recommend removing it from the current PR

File: config/credentials.yml.enc [hunk 2]
```diff
@@ -1 +1 @@
-fQ2jKfPahjJVgDaMoii60oltc9L4H6+0DfsoajMHkO69fSRqdITIWLl/BAdTZqUpgmEONQ4lUnfHieefDm2lasDQfCKFjsQm2+Vo8Q0n1nEDhedM+9yUnzwznJ4xG2o5x+14VZxhTp9l7XUM/4AMRLWc5R9kKnzvEhBK2YUbqpMHtfnoXThYrhqvh3jB+Q2SN49CrnZ8Wen+c4XWMSOfeVFWc0/zQgnJBcc3M/1cAtr8HpwHY6M0pd0jKAQcQV6hBfO8tf6HORF8tu/Aftc4p0o7dt5O5/dYBSg2f0OkW75kUV34o+tCwy0o/CuHLrIyIa8IB2kHa5gnY5eRtJ6VwjhfvWVF1umh/8f/7eQfroZeSVf52ZJnjbUKRRnjTWL603b6RJFWmocPjtUzGvYmjaLllPK44ScAqO9p--e6GFzr9G+9V+EG8I--YPfKFko0vkqw3cCErsVA0A==
\ No newline at end of file
+cIkw9nbMrBnKtRMRE2U0nhu3Gn/aRet9r6a3/9/FiANquiebJmaq+/CnNiBrpZPI8AeU8DihDLlWDd2rD+TcTmIMaplx6EzZmOfy1EF7m4wfuMWbP4LbtB81P+fEhgZMwiy01vpJTVLPvrI3ulfpXNTpxyWtCFvhagKqP+NSEcpt2OXm+LlxjCaG+hCmLrAj7djqqfPygJne4VdFmmGKf93AYAExAYS86IDqu7ol+zQrY67lKlcuRCGhhmknjykAxdFrTStD6Acwh81HpeCrkRcm3i/fODVBOWoUPcySu32HrOaeexlM/DpervD9bQYgnUsFDQ+oLUhlU05eNt2JqphH4N/tcf+nssrgNkpWh12HKBmZRwnxD+clWCYXMRTHgoKxt1iBdOrUmTBPzKRYXWdiC5mvveLMtte2--LciPAg/zlLo0QbIk--RHXNaK9TQg6pG0kH7T2k3g==
```
Comment (hannazspot) [inline config/credentials.yml.enc]: Why is it changed?

File: spec/api/v1/admin/categories_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe V1::Admin::Categories do
```
Comment (hannazspot) [inline spec/api/v1/admin/categories_spec.rb:3]: I'd recommend using `describe V1::Admin::Categories, type: :controller` here

File: spec/api/v1/admin/categories_spec.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  describe 'POST /api/v1/admin/categories' do
+    let(:params) do
+      {
+        title: 'nurse',
+        description: 'medical nursing'
+      }
+    end
+
+    it 'returns created category' do
+      post '/api/v1/admin/categories', params: params, headers: headers(current_user)
```
Comment (hannazspot) [inline spec/api/v1/admin/categories_spec.rb:17]: if you use the `controller` type you'll be able to do `post :categories`

**Review by hannazspot — APPROVED** [review]

**Review by PandaGrom — APPROVED** [review]

**Pull request merged**